src/dmd/traits.d

```diff
@@ -444,6 +445,52 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         return new IntegerExp(e.loc, false, Type.tbool);
     }
 
+    /**
+       Gets a function type from a given AST node 
+       if the node is a function of some sort.
+
+     Params:
+        o = an AST node to check for a `TypeFunction`  
+        fdp = optional pointer to a function declararion, to be set 
+          if `o` is a function declarartion.
+
+     Returns: 
+        a type node if `o` is a declaration of
+            a delegate, function, function-pointer
+          or a variable of the former.  Otherwise, `null`.
+    */
+
+    static TypeFunction traitsFuncArg(RootObject o, FuncDeclaration* fdp = null)
```

Comment:
All other DDoc headers in DMD seem to follow this convention:

```D
/*************************************************************
 * Whatever
 * Whatever
 */
```

Please remove the extra space between the DDoc header and the function declaration.

I cannot, in good conscience, approve a function name like `traitsFuncArg`.  Please use a more descriptive name like `getFuncType` or something along those lines (Yes, I know my own names have been poor).

Reply:
> Please remove the extra space between the DDoc header and the function declaration.

Hmm the above is the Phobos style:
Compare with e.g. https://github.com/dlang/phobos/blob/master/std/algorithm/comparison.d

If you really want to stick with this style at DMD, then this should be documented in a DMD-specific style guide?

Reply:
> Hmm the above is the Phobos style.

I see.

> If you really want to stick with this style at DMD, then this should be documented in a DMD-specific style guide.

Agreed.  

Aside from the name then, LGTM.

Reply:
Also, consider making this a non-member function instead of static.  Refer to [@WalterBright's comment](https://github.com/dlang/dmd/pull/7405#pullrequestreview-82322025)

Reply:
Phobos has a mixed style when it comes to comments [1] [2] [3].

[1] https://github.com/dlang/phobos/blob/master/std/string.d#L2194-L2205
[2] https://github.com/dlang/phobos/blob/master/std/string.d#L2254-L2267
[3] https://github.com/dlang/phobos/blob/master/std/string.d#L2332-L2358

Reply:
Let's use https://github.com/dlang/phobos/blob/master/std/string.d#L2254-L2267 for new code, thanks.